docs/docs/tutorial/chapter3/saving-data.md

```diff
@@ -297,35 +440,39 @@ const ContactPage = () => {
 export default ContactPage
 ```
 
-We reference the `createContact` mutation we defined in the Contacts SDL passing it an `input` object which will contain the actual name, email and message values.
-
-Next we'll call the `useMutation` hook provided by Redwood which will allow us to execute the mutation when we're ready (don't forget to `import` it):
+</TabItem>
+<TabItem value="ts" label="TypeScript">
 
-```jsx title="web/src/pages/ContactPage/ContactPage.js"
-// highlight-next-line
-import { MetaTags, useMutation } from '@redwoodjs/web'
+```tsx title="web/src/pages/ContactPage/ContactPage.tsx"
+import { MetaTags } from '@redwoodjs/web'
 import {
   FieldError,
   Form,
   Label,
   TextField,
   TextAreaField,
   Submit,
+  SubmitHandler,
 } from '@redwoodjs/forms'
 
+// highlight-start
 const CREATE_CONTACT = gql`
   mutation CreateContactMutation($input: CreateContactInput!) {
     createContact(input: $input) {
       id
     }
   }
 `
+// highlight-end
 
-const ContactPage = () => {
-  // highlight-next-line
-  const [create] = useMutation(CREATE_CONTACT)
+interface FormValues {
+  name: string
+  email: string
+  message: string
+}
 
-  const onSubmit = (data) => {
+const ContactPage = () => {
+  const onSubmit: SubmitHandler<FormValues> = (data) => {
```

Comment:
```suggestion
  const onSubmit: SubmitHandler<FormValues> = (data: CreateContactInput) => {
```

Reply:
`data` is already typed thanks to `SubmitHandler<FormValues>`